org.eclipse.jdt.core/compiler/org/eclipse/jdt/internal/compiler/ast/LambdaExpression.java

```diff
@@ -468,7 +480,7 @@ else if (this.expectedType instanceof ReferenceBinding)
 		if ((this.binding.tagBits & TagBits.HasMissingType) != 0) {
 			this.scope.problemReporter().missingTypeInLambda(this, this.binding);
 		}
-		if (this.shouldCaptureInstance && this.scope.isConstructorCall) {
```

Comment:
Trying to understand the rationale of this addition, I set a breakpoint with condition `this.shouldCaptureInstance && this.scope.isConstructorCall && this.hasOuterClassMemberReference` , i.e, just the new part negated, but in the lambda-related tests I couldn't observe this breakpoint triggering.

I suspect this addition has been obsoleted by the addition in Scope.java:5463 , where we now avoid setting `shouldCaptureInstance` in this particular situation.